packages/backend/routes/builders.js

```diff
@@ -238,6 +238,12 @@ router.post("/set-batch-number", withAddress, async (req, res) => {
     batch,
   };
 
+  const user = await db.findUserByAddress(address);
+  if (user.data.batch.number) {
+    res.status(401).send("🚫 User already in batch");
+    return;
+  }
```

Comment:
This will throw a Typerror if the batch is not defined. you can do `user.data?.batch?.number`.... or you can remove the first `?` if you check if `user.exists` first